### PR TITLE
Import a molecule to the server by an identifier

### PR DESCRIPTION
**Pull request by psavery**
A molecule can be imported by its inchikey, inchi,
or smiles.

Signed-off-by: Patrick Avery <patrick.avery@kitware.com>

### CODE REVIEW

**Comment by psavery** [comment]
@cjh1 I haven't checked anything about the molecule, yet, but I was able to import a molecule like this:

```
import openchemistry as oc
  
# Search for ethanol. Make sure it is in the database first.
smiles = 'CCO'

molecule = oc.import_structure(smiles=smiles)

print('molecule is', molecule)
# molecule is <openchemistry.GirderMolecule object at 0x7fc6c3829dd8>
```

**Comment by psavery** [comment]
Also, importing by SMILES only works with [this PR](https://github.com/OpenChemistry/mongochemserver/pull/102) on mongochemserver.

File: openchemistry/__init__.py [hunk 1]
```diff
@@ -668,6 +668,32 @@ def _find_using_cactus(identifier):
...
+    if inchi:
+        params['inchi'] = inchi
+    if smiles:
+        params['smiles'] = smiles
+
+    if not params:
+        raise Exception('At least one of these must be set: '
+                        'inchikey, inchi, or smiles')
+
+    molecules = girder_client.get('molecules', parameters=params)
```
Comment (cjh1) [inline openchemistry/__init__.py]: Can we use the POST ( create endpoint? ) here?

Comment (psavery) [reply openchemistry/__init__.py]: Oh yes... I have that in my notes. Sorry I missed that.

Comment (psavery) [reply openchemistry/__init__.py]: It is fixed. Now, if the molecule doesn't exist in the database, one will be created.

**Review by cjh1 — APPROVED** [review]

**Comment by cjh1** [comment]
@psavery As a follow up PR lets add `smiles` and `inchikey` to the parameter list for `find_structure`

**Comment by psavery** [comment]
Okay, will do!

**Pull request merged**